### ISSUE TITLE
Feature Request: Auto-focus on First Field with Error in Form Validation

### ISSUE THREAD

**Issue by romanpalenik** [ant-design/ant-design#45701]
#### Issue Description

While the scrollToFirstError property provides a smooth scroll to the first field that encounters a validation error, it does not automatically set focus to this field. This can be a hindrance to accessibility, as users who rely on keyboard navigation will not automatically know which field has the error, especially after the page scrolls. Moreover, for screen reader users, the error message might not be read out if the focus does not shift to the input with the error.

#### Current Behavior

The form automatically scrolls to the first field with an error, but it does not change the focus to that field.
Expected Behavior

When a form validation fails:

    The page should scroll to the first invalid input, as it currently does.
    The focus should automatically move to that field.
    The input should have aria-describedby set to the ID of the error message so that the error is announced when the field is focused.

This behavior will not only improve accessibility but will also enhance the user experience for everyone by clearly indicating which field needs attention.

### Feature Proposal

I propose an enhancement to the scrollToFirstError option within the Form component. This enhancement would allow developers to specify whether the focus should be set to the first field with an error after scrolling.

### What does the proposed API look like?

`<Form scrollToFirstError={{ behavior: 'smooth', block: 'center', focus: true }}>
`

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/pull/51231
